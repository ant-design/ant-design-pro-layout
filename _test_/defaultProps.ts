export default {
  route: {
    path: '/',
    routes: [
      {
        path: '/',
        name: 'welcome',
        icon: 'smile',
        routes: [
          {
            path: '/welcome',
            name: 'one',
            icon: '/favicon.png',
            routes: [
              {
                path: '/welcome/welcome',
                name: 'two',
                icon: 'smile',
                exact: true,
              },
            ],
          },
        ],
      },
    ],
  },
  location: {
    pathname: '/',
  },
};
