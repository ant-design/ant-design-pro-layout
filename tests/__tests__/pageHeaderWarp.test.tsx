import { render, mount } from 'enzyme';
import React from 'react';
import ProLayout, { PageHeaderWrapper } from '../../src';
import defaultProps from './defaultProps';
import { waitForComponentToPaint } from './util';

describe('BasicLayout', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => ({
        matches: false,
        addListener() {},
        removeListener() {},
      })),
    });
    Object.defineProperty(window, 'localStorage', {
      value: {
        getItem: jest.fn(() => 'zh-CN'),
      },
    });
  });

  it('base use', () => {
    const html = render(
      <ProLayout {...defaultProps}>
        <PageHeaderWrapper />
      </ProLayout>,
    );
    expect(html).toMatchSnapshot();
  });

  it('content is text', () => {
    const html = render(
      <ProLayout {...defaultProps}>
        <PageHeaderWrapper content="just so so" />
      </ProLayout>,
    );
    expect(html).toMatchSnapshot();
  });

  it('title=false, don not render title view', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageHeaderWrapper title={false} />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-page-header-heading-title')).toHaveLength(0);
  });

  it('have default title', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageHeaderWrapper />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const titleDom = wrapper.find('.ant-page-header-heading-title');
    expect(titleDom.text()).toEqual('welcome');
  });

  it('title overrides the default title', async () => {
    const wrapper = mount(
      <ProLayout {...defaultProps}>
        <PageHeaderWrapper title="name" />
      </ProLayout>,
    );
    await waitForComponentToPaint(wrapper);
    const titleDom = wrapper.find('.ant-page-header-heading-title');
    expect(titleDom.text()).toEqual('name');
  });
});
