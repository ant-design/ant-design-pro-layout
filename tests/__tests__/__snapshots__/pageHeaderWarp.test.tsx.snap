// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout base use 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="height:100%"
  >
    <aside
      class="ant-pro-sider-menu-sider ant-layout-sider ant-layout-sider-dark"
      style="flex:0 0 256px;max-width:256px;min-width:256px;width:256px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-menu-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header"
        style="padding:0;width:100%;z-index:9"
      >
        <div
          class="ant-pro-global-header"
        >
          <span
            class="ant-pro-global-header-trigger"
          >
            <i
              aria-label="icon: menu-fold"
              class="anticon anticon-menu-fold"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="menu-fold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
                />
              </svg>
            </i>
          </span>
        </div>
      </header>
      <main
        class="ant-pro-basicLayout-content ant-pro-basicLayout-has-header ant-layout-content"
        style="min-height:100%"
      >
        <div>
          <div
            class="ant-pro-page-header-wrap"
          >
            <div
              class="ant-pro-page-header-wrap-page-header-warp"
            >
              <div
                class="ant-pro-grid-content"
              >
                <div
                  class="ant-page-header ant-page-header-ghost"
                >
                  <div
                    class="ant-page-header-heading"
                  >
                    <span
                      class="ant-page-header-heading-title"
                    >
                      welcome
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</div>
`;

exports[`BasicLayout content is text 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="height:100%"
  >
    <aside
      class="ant-pro-sider-menu-sider ant-layout-sider ant-layout-sider-dark"
      style="flex:0 0 256px;max-width:256px;min-width:256px;width:256px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-menu-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header"
        style="padding:0;width:100%;z-index:9"
      >
        <div
          class="ant-pro-global-header"
        >
          <span
            class="ant-pro-global-header-trigger"
          >
            <i
              aria-label="icon: menu-fold"
              class="anticon anticon-menu-fold"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="menu-fold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
                />
              </svg>
            </i>
          </span>
        </div>
      </header>
      <main
        class="ant-pro-basicLayout-content ant-pro-basicLayout-has-header ant-layout-content"
        style="min-height:100%"
      >
        <div>
          <div
            class="ant-pro-page-header-wrap"
          >
            <div
              class="ant-pro-page-header-wrap-page-header-warp"
            >
              <div
                class="ant-pro-grid-content"
              >
                <div
                  class="ant-page-header ant-page-header-ghost"
                >
                  <div
                    class="ant-page-header-heading"
                  >
                    <span
                      class="ant-page-header-heading-title"
                    >
                      welcome
                    </span>
                  </div>
                  <div
                    class="ant-page-header-content"
                  >
                    <div
                      class="ant-pro-page-header-wrap-detail"
                    >
                      <div
                        class="ant-pro-page-header-wrap-main"
                      >
                        <div
                          class="ant-pro-page-header-wrap-row"
                        >
                          <div
                            class="ant-pro-page-header-wrap-content"
                          >
                            just so so
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</div>
`;
